src/Entity/Report.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Silktide\ProspectClient\Entity;
+
+class Report extends AbstractEntity
+{
+    public function getId(): string
+    {
+        return $this->jsonData->{"report_id"};
+    }
+
+    public function getAccountId(): string
+    {
+        return $this->jsonData->{"account_id"};
+    }
+
+    public function getDomain(): string
+    {
+        return $this->jsonData->{"domain"};
+    }
+
+    public function getOverallScore(): int
+    {
+        return $this->jsonData->{"overall_score"};
+    }
+
+    /**
+     * @return object|null A plain PHP object with properties of the requested section, or null if
+     * the section does not exist.
+     */
+    public function getReportSection(string $name): ?object
```

Comment:
I'm not really sure about this function. If we look at an example response:
```
{
   "domain":"silktide.com",
   "contact_details":{
      "email":false
   },
   "local_presence":{
      "detected_phone":"+44 1322 460460",
      "detected_address":"Silktide LTD, Brunel Parkway, Pride Park, DE24 8HR, United Kingdom, United Kingdom",
      "detected_name":"Silktide"
   },
   "facebook_page":{
      "page_link":"https:\/\/www.facebook.com\/silktide\/",
      "page_likes":46560
   },
   "organic_search":{
      "average_monthly_traffic":5835
   },
   "mobile":{
      "is_mobile":true,
      "is_tablet":true
   },
   "analytics":{
      "analytics_tool":"Google Analytics"
   },
   "incoming_links":{
      "total_backlinks":156152
   },
   "social_sharing":{
      "url_likes":14
   },
   "domain_age":{
      "domain_age_days":5560
   },
   "page_count":{
      "pages_discovered_count":31
   },
   "video":{
      "has_video":false
   },
   "amount_of_content":{
      "total_word_count":1295,
      "average_words_per_page":259
   },
   "last_updated":{
      "days_since_update":108
   },
   "website_speed":{
      "average_homepage_load_time_seconds":4.934
   },
   "reviews":{
      "reviews_found_count":1,
      "probably_more_reviews":false
   },
   "server_behaviour":{
      "uses_gzip_compression":true,
      "error_page_sends_404":true
   },
   "page_titles_and_descriptions":{
      "homepage_title_tag":"Silktide - making the web a better place",
      "homepage_meta_description":"Helping to make the web a better place.",
      "pages_missing_title_count":0,
      "pages_missing_description_count":0,
      "pages_duplicated_title_count":0,
      "pages_duplicated_description_count":0
   },
   "report_id":"7b8cf004712790702635a495d6e1bf1572f67c7a",
   "account_id":"prospect_testing",
   "meta":{
      "analysis_country":"GB",
      "started_processing_at":"2016-05-27T10:04:43+00:00",
      "report_requested_at":"2016-05-27T10:04:29+00:00",
      "report_completed_at":"2016-05-27T10:05:08+0000",
      "requested_by":"someguy@silktide.com"
   },
   "overall_score":83,
   "analysed_page_count":5
}
```

It would seem that a common thing you might want is to return something like reviews (yay), but we currently provide no provision for someone wanting to get say, the analysed_page_count.

Someone might try with this function, but it'd throw a typehint error, and it would feel like asking about a section would be the incorrect term anyway

(Also, section? We seem to use this on the docs, but I thought these were tests that were enabled. Have we started outwardly calling these sections or are we just bad on our microcopy @andywaite ?)

We should also probably be providing the user with a function that allows them to get the raw underlying array, or at least a list of all the "sections"

Reply:
:+1: This needs attention, as I wrongly assumed everything returned was an object.